src/algebra/ring/basic.lean

```diff
@@ -1168,3 +1168,99 @@ theorem neg_left : commute a b → commute (- a) b := semiconj_by.neg_left
 @[simp] theorem sub_left : commute a c → commute b c → commute (a - b) c := semiconj_by.sub_left
 
 end commute
+section prop_typeclasses
+
+/-! ### Propositional typeclasses on several ring equivalences, useful in the theory of semilinear
+maps
+
+-/
+variables {R₁ : Type*} {R₂ : Type*} {R₃ : Type*}
+variables [semiring R₁] [semiring R₂] [semiring R₃]
+variables (σ₁₂ : R₁ →+* R₂) (σ₂₃ : R₂ →+* R₃) (σ₁₃ : out_param (R₁ →+* R₃))
```

Comment:
Is `out_param` meaningful on lemmas or instances? I feel like this should probably be limited to just the `class` definition.

Reply:
You're right -- I limited the `out_param` to the class definitions.